props-core/src/main/java/sh/props/tuples/Pair.java

```diff
@@ -55,21 +56,35 @@
   /**
    * Constructs a new {@link Pair} with the updated value.
    *
+   * @param pair the original pair that will provide the other values
    * @param value the new value to set
+   * @param <T> the type of the first object
+   * @param <U> the type of the second object
    * @return a new object with the value updated
    */
-  public Pair<T, U> updateFirst(@Nullable T value) {
-    return new Pair<>(value, this.second);
+  public static <T, U> Pair<T, U> updateFirst(@Nullable Pair<T, U> pair, @Nullable T value) {
+    if (pair == null) {
+      return new Pair<>(value, null);
+    }
+
+    return new Pair<>(value, pair.second);
   }
 
   /**
    * Constructs a new {@link Pair} with the updated value.
    *
+   * @param pair the original pair that will provide the other values
    * @param value the new value to set
+   * @param <T> the type of the first object
+   * @param <U> the type of the second object
    * @return a new object with the value updated
    */
-  public Pair<T, U> updateSecond(@Nullable U value) {
-    return new Pair<>(this.first, value);
+  public static <T, U> Pair<T, U> updateSecond(@Nullable Pair<T, U> pair, @Nullable U value) {
```

Comment:
it's more useful for this method to be a static, since we are not always in a situation where an original tuple object exists.  That way, we can call the static and avoid NPE checks in calling code.